lib/src/form_builder_validators.dart

```diff
@@ -201,25 +201,28 @@ class FormBuilderValidators {
     };
   }
 
+  // TODO(any): l10n
   /// [FormFieldValidator] that requires the field's value to be a valid integer.
-  static FormFieldValidator integer({
+  static FormFieldValidator integer(
+    BuildContext context, {
     String errorText = 'Value must be an integer.',
   }) {
     return (valueCandidate) {
-      if (int.tryParse(valueCandidate) == null && valueCandidate.isNotEmpty) {
+      if (int.tryParse(valueCandidate ?? '') == null) {
         return errorText;
       }
       return null;
     };
   }
 
+  // TODO(any): l10n
   /// [FormFieldValidator] that requires the field's value to be a valid double.
-  static FormFieldValidator double_({
+  static FormFieldValidator double_(
+    BuildContext context, {
     String errorText = 'Value must be a double.',
   }) {
     return (valueCandidate) {
-      if (double.tryParse(valueCandidate) == null &&
-          valueCandidate.isNotEmpty) {
+      if (double.tryParse(valueCandidate ?? '') == null) {
```

Comment:
Is this really the desired behavior?  This way, the value can't be empty.  But shouldn't this validation simply validate that the value is a valid double -- and if you want to make it required, you would also add a `required()` validator?

I'd recommend:
```
  return valueCandidate.isNotEmpty && double.tryParse(valueCandidate) == null
    ? errorText
    : null;

```

Reply:
I based myself on the 'num' validator already present, but I agree with you for the behavior and the ternary!

Reply:
Ternary aside, my main point is that an empty value should not trigger an error.  I just checked `master` and that is how `numeric` behaves:
```dart
  /// [FormFieldValidator] that requires the field's value to be a valid number.
  static FormFieldValidator numeric({
    String errorText = 'Value must be numeric.',
  }) {
    return (valueCandidate) {
      if (num.tryParse(valueCandidate) == null && valueCandidate.isNotEmpty) {
        return errorText;
      }
      return null;
    };
  }
```

I prefer checking the string is not empty first because then you don't need the `??  ''` expression.

Reply:
So the codes becomes

```dart
static FormFieldValidator integer(
  BuildContext context, {
  String errorText,
  int radix,
}) {
  return (valueCandidate) {
    if (valueCandidate.isNotEmpty &&
        int.tryParse(valueCandidate, radix: radix) == null) {
      return errorText ??
          FormBuilderLocalizations.of(context).numericErrorText;
    }
    return null;
 };
}
```

Reply:
LGTM.  Note that a non-numeric input like, 'ABC' would yield a _null_.  I think that's OK -- the user needs to do something else to ensure numeric input.  (I like overriding the input field.)

Reply:
I am mitigated, because ABC should return an error message too I think, but let's start already with this version!

Reply:
But it is true that the developer has to make sure that the textfield is properly configured in numeric mode.